### PR TITLE
Make a tip files more markdown way

### PR DESCRIPTION
**Pull request by micpiotrowski**

### CODE REVIEW

File: src/app/tips.service.ts [hunk 1]
```diff
@@ -26,14 +26,13 @@ export class TipsService {
...
-    const splittedContent = content.split(this.tipsSeparator);
+    const splittedContent = content.split(this.tipsSeparator).slice(1);
     const tips: Tips = [];
     for(let singleTipContent of splittedContent) {
-      const splittedSingleTipContent = singleTipContent.split(/\n/);
       const tip: Tip = {
-        name: splittedSingleTipContent[0],
-        date: splittedSingleTipContent[1],
-        content: splittedSingleTipContent.slice(3).join("\n")
+        name: singleTipContent.match(/^\s*(.*)\s*\n/)[1],
```
Comment (PawelLipski) [inline src/app/tips.service.ts:33]: Hmmm just thinking if this can be done in a less opaque way...

Comment (PawelLipski) [reply src/app/tips.service.ts:33]: Are you currently using some JS library for converting markdown to HTML? :thinking:

File: src/app/tips.service.ts [hunk 2]
```diff
@@ -26,14 +26,13 @@ export class TipsService {
...
+        date: singleTipContent.match(/^###\s(.*)\s*$/m)[1],
+        content: singleTipContent.split(/\n/).slice(1).filter(string => !string.match(/^###\s+/)).join("\n")
```
Comment (PawelLipski) [inline src/app/tips.service.ts:35]: TBH the previous way with `====` seemed less brittle than the current approach :/

File: src/app/tips.service.ts (hunk 1 above)
Comment (micpiotrowski) [reply src/app/tips.service.ts:33]: Yes, I use a library but for clarity and flexibility I can't use it to simply convert whole MD file to a HTML. I'd rather extract a MD file content to a JS structure (like array of `Tip` objects here) and then use that structure in a future conversion. E.g. this approach give us a possibility to sort a tips descending by a date (the newest will be the first) regardless of tip order in a MD file (I'm working on it).

File: src/app/tips.service.ts (hunk 2 above)
Comment (micpiotrowski) [reply src/app/tips.service.ts:35]: Why? It's simply skipping a first line and lines that starts with `/^###\s+/` (the date line)

Comment (micpiotrowski) [reply src/app/tips.service.ts:35]: Ouh, ok, you are mean a `/=======\n/` separator

Comment (micpiotrowski) [reply src/app/tips.service.ts:35]: But now, new tips separator is simply a 2nd size header `/^## /gm`.
I just make an assumption that every new tip begin with a title in a `## Header`

Comment (PawelLipski) [reply src/app/tips.service.ts:35]: Ok, let's leave as is then

**Review by PawelLipski — APPROVED** [review]

**Pull request merged**